### PR TITLE
Bump testcontainers version

### PR DESCRIPTION
**Pull request by geoand**
Needed to fix https://github.com/quarkusio/quarkus/issues/6588#issuecomment-623722539

### CODE REVIEW

File: getting-started/pom.xml [hunk 1]
```diff
@@ -46,6 +46,22 @@
...
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-container-image-jib</artifactId>
+        </dependency>
+
+<!--        <dependency>-->
+<!--            <groupId>io.quarkus</groupId>-->
+<!--            <artifactId>quarkus-minikube-deployment</artifactId>-->
+<!--            <scope>provided</scope>-->
+<!--            <version>${quarkus-plugin.version}</version>-->
+<!--        </dependency>-->
```
Comment (gsmet) [inline getting-started/pom.xml:64]: This doesn't look like something you want to merge :).

File: neo4j-quickstart/pom.xml [hunk 2]
```diff
@@ -15,7 +15,7 @@
     <maven.compiler.source>1.8</maven.compiler.source>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.target>1.8</maven.compiler.target>
-    <testcontainers.version>1.12.5</testcontainers.version>
+    <testcontainers.version>1.14.1</testcontainers.version>
```
Comment (gsmet) [inline neo4j-quickstart/pom.xml:18]: It's in the BOM so maybe we should just drop the version?

File: getting-started/pom.xml (hunk 1 above)
Comment (geoand) [reply getting-started/pom.xml:64]: Absolutely not :)

File: neo4j-quickstart/pom.xml (hunk 2 above)
Comment (geoand) [reply neo4j-quickstart/pom.xml:18]: Yeah, I saw the BOM had testcontainers after I made the change. I'll update

File: getting-started/pom.xml [hunk 3]
```diff
@@ -46,6 +46,22 @@
             <artifactId>rest-assured</artifactId>
             <scope>test</scope>
         </dependency>
+
```
Comment (cescoffier) [inline getting-started/pom.xml:49]: Can it be removed?
If not, can a comment be added to explain why this dependency is there but commented.

File: getting-started/pom.xml [hunk 4]
```diff
@@ -46,6 +46,22 @@
...
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-container-image-jib</artifactId>
+        </dependency>
+
+<!--        <dependency>-->
+<!--            <groupId>io.quarkus</groupId>-->
+<!--            <artifactId>quarkus-minikube-deployment</artifactId>-->
+<!--            <scope>provided</scope>-->
+<!--            <version>${quarkus-plugin.version}</version>-->
```
Comment (cescoffier) [inline getting-started/pom.xml:63]: Same question for this dependency.
In addition, I’m concerned by the version.

File: neo4j-quickstart/pom.xml (hunk 2 above)
Comment (geoand) [reply neo4j-quickstart/pom.xml:18]: Actually, it's not in the BOM, yet :wink: https://github.com/quarkusio/quarkus/pull/9073

**Comment by geoand** [comment]
I added neo4j from testcontainers to the BOM in quarkusio/quarkus#9073.

Once that is in, I'll update this PR to remove the version and fix the extra stuff that slipped in.

**Comment by geoand** [comment]
Issues should be addressed

**Comment by geoand** [comment]
PR was updated to add testcontainers.version to all quickstarts that use TestContainers since the BOM no longer enforces the version.

**Comment by cescoffier** [comment]
This should be marked as a breaking change as many projects rely on these versions. At least, I have a few projects that do depend on them.

**Comment by geoand** [comment]
Yes, we have already marked the PR that did the change as breaking

**Comment by gsmet** [comment]
I even added a note in the migration guide yesterday :).

**Comment by geoand** [comment]
@cescoffier OK to merge?

**Comment by cescoffier** [comment]
Done!

**Comment by geoand** [comment]
Excellent, thanks!

**Pull request merged**